core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -4004,6 +4471,99 @@ class KafkaApis(val requestChannel: RequestChannel,
     CompletableFuture.completedFuture[Unit](())
   }
 
+
+  def getAcknowledgeBatchesFromShareFetchRequest(
+                                                  shareFetchRequest : ShareFetchRequest,
+                                                  topicNames : util.Map[Uuid, String],
+                                                  erroneous : mutable.Map[TopicIdPartition, ShareAcknowledgeResponseData.PartitionData],
+                                                ) : mutable.Map[TopicIdPartition, util.List[ShareAcknowledgementBatch]] = {
+
+    val acknowledgeBatchesMap = mutable.Map[TopicIdPartition, util.List[ShareAcknowledgementBatch]]()
+    shareFetchRequest.data().topics().forEach ( topic => {
+
+      if(!topicNames.asScala.contains(topic.topicId)) {
```

Comment:
I am not sure how costly the API for `asScala` on Map is, but should have some cost, so do you want to have conversion in forEack loop?